pytest_mpi/parallel_assert.py

```diff
@@ -19,24 +19,37 @@ def parallel_assert(assertion: Callable, participating: bool = True, msg: str =
     Notes
     -----
     It is very important that ``parallel_assert`` is called collectively on all
-    ranks simulataneously.
+    ranks simultaneously.
+    This function allows passing a callable instead of a boolean for the
+    `assertion` argument. This is useful in rare circumstances, such as when
+    the assertion is not defined on all tasks, but is not recommended.
+
 
     Example
     -------
     Where in serial code one would have previously written:
     ```python
     x = f()
-    assert x < 5
+    assert x < 5, "x is too large"
     ```
 
     Now write:
     ```python
     x = f()
-    parallel_assert(lambda: x < 5)
+    parallel_assert(x < 5, "x is too large")
```

Comment:
Ahhh. This is a good API spot. We should have
```py
def parallel_assert(assertion, msg="", *, participating):
    ...
```
 as the function signature. I think that that is fine to change because I don't think anyone is using `participating` without setting as a kwarg.